Store image digest in image build tasks

concaf opened the pull request:
This commit makes the tasks under `task/` directory store the digest of
the image built during task execution; this gives developers access to
the image sha of the built images for further use.

vdemeester commented on task/buildah-pr/0.1/buildah-pr.yaml:
```diff
@@ -42,11 +45,13 @@ spec:
       - name: image
         type: image
-
+  workspaces:
+    - name: source
+      mountPath: /workspace/source
```
for the `-pr` task, we don't use workspaces, on pipeline resource

vdemeester commented on task/buildah-pr/0.1/buildah-pr.yaml:
```diff
@@ -42,11 +45,13 @@ spec:
       - name: image
         type: image
-
+  workspaces:
+    - name: source
+      mountPath: /workspace/source
```
for the `-pr` task, we don't use workspaces, on pipeline resource

concaf replied:
Fixed

vdemeester commented on task/buildah-pr/0.1/buildah-pr.yaml:
```diff
@@ -42,11 +45,13 @@ spec:
     - name: build
       image: $(params.BUILDER_IMAGE)
-      workingDir: /workspace/source
+      workingDir: $(workspaces.source.path)
```
This shouldn't be changed in the `-pr` task. (or maybe using `$(resources.inputs.source.path)` if it works)

concaf replied:
Done, replaced with `$(resources.inputs.source.path)`

vdemeester commented on task/buildah-pr/0.1/buildah-pr.yaml:
```diff
@@ -56,14 +61,17 @@ spec:
       image: $(params.BUILDER_IMAGE)
-      workingDir: /workspace/source
-      command: ['buildah', 'push', '--tls-verify=$(params.TLSVERIFY)', '$(resources.outputs.image.url)', 'docker://$(resources.outputs.image.url)']
+      workingDir: $(workspaces.source.path)
```
Same here, this shouldn't be changed in the `-pr` task. (or maybe using `$(resources.inputs.source.path)` if it works)

concaf replied:
Done, replaced with `$(resources.inputs.source.path)`

vdemeester reviewed (changes requested):
@concaf thanks for the PR 👍🏼 As per my comments (below), we shouldn't add workspaces to `-pr` tasks though (`pr` in the name stands for PipelineResource) and for those, we should only rely on the resources inputs

concaf committed:
Store image digest in image build tasks

This commit makes the tasks under `task/` directory store the digest of
the image built during task execution; this gives developers access to
the image sha of the built images for further use.

concaf commented:
@vdemeester thanks for the review. I could get the `-pr` resources working with `$(resources.inputs.source.path)` instead of workspaces. Let me know if you need something else.

chmouel commented:
Any chances to do a rebase with my last change? that would properly test the catalog

chmouel commented:
the testing of this repo is pretty brand new so sorry for the extra step, can you do a :

`make generate`

commit it ? this will apply the change you made and generate the [test pipeline](https://github.com/openshift/pipelines-catalog/blob/master/.tekton/pipeline.yaml#L2064) so tekton-asa-code can pick it up.

i need to fix this check https://github.com/openshift/pipelines-catalog/blob/master/openshift/gen/prestep.yaml#L56-L75 since it's not working

chmouel commented:
/hold

concaf committed:
Add generated files

This commit includes files generated using `make generate`

chmouel commented:
/hold cancel
/lgtm

Looking Good! 🤙🏽
